substrate/frame/stake-tracker/src/tests.rs

```diff
@@ -0,0 +1,144 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg(test)]
+
+use crate::mock::*;
+
+use frame_election_provider_support::SortedListProvider;
+use sp_staking::Stake;
+
+#[test]
```

Comment:
one goal of having this pallet be a separate entity is to have it have more extensive tests that cover edge cases beyond what would be the common scenarios when it is coupled with staking. High level, this does not seem to be achieved here.